dds/DCPS/Serializer.cpp

```diff
@@ -196,6 +196,11 @@ bool EncapsulationHeader::to_encoding(
   return true;
 }
 
+void EncapsulationHeader::set_encapsulation_options(Message_Block_Ptr& mb)
+{
+  mb->rd_ptr()[padding_marker_byte_index] |= ((padding_marker_alignmet - mb->length() % padding_marker_alignmet) & 0x03);
```

Comment:
Consider adding a check to make sure `mb` has sufficient length.

Add a test of this function to `tests/DCPS/Serializer`

Reply:
What should we use as a test criteria in the test function?

Reply:
I think that would be based on the requirements in the spec.